src/main/java/org/zalando/planb/revocation/persistence/CassandraAuthorizationRuleStore.java

```diff
@@ -0,0 +1,112 @@
+package org.zalando.planb.revocation.persistence;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.RegularStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.zalando.planb.revocation.domain.AuthorizationRule;
+import org.zalando.planb.revocation.domain.ImmutableAuthorizationRule;
+
+import javax.annotation.PostConstruct;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.datastax.driver.core.querybuilder.QueryBuilder.bindMarker;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.now;
+
+@Slf4j
+public class CassandraAuthorizationRuleStore implements AuthorizationRulesStore.Internal {
+
+    private static final long RELOAD_IN_MS =  1000 * 60 * 10; // 10 minutes
+
+    private List<AuthorizationRule> inMemoryRuleStore = Collections.emptyList();
+
+    private final Session session;
+
+    private final PreparedStatement getRules;
+
+    private final PreparedStatement insertRule;
+
+    private static final String AUTHORIZATION_TABLE = "authorization_rule";
+    private static final String REQUIRED_USER_CLAIMS = "required_user_claims";
+    private static final String ALLOWED_REVOCATION_CLAIMS = "allowed_revocation_claims";
+    private static final String CREATED_BY = "created_by";
+    private static final String LAST_MODIFIED_BY = "last_modified_by";
+    private static final String UUID = "uuid";
+
+    private static final RegularStatement INSERT_AUTHORIZATION = QueryBuilder.insertInto(AUTHORIZATION_TABLE)
+            .value(UUID, now())
+            .value(REQUIRED_USER_CLAIMS, bindMarker())
+            .value(ALLOWED_REVOCATION_CLAIMS, bindMarker())
+            .value(CREATED_BY, bindMarker())
+            .value(LAST_MODIFIED_BY, bindMarker());
+    private static final RegularStatement SELECT_AUTHORIZATION = QueryBuilder.select()
+            .column(REQUIRED_USER_CLAIMS)
+            .column(ALLOWED_REVOCATION_CLAIMS)
+            .from(AUTHORIZATION_TABLE);
+    private static final RegularStatement CLEANUP_AUTHORIZATION = QueryBuilder.truncate(AUTHORIZATION_TABLE);
```

Comment:
why do we have a "TRUNCATE" here? is this only called from tests or is there a real use case?